DDOT_module/child_classes/correction_data_generator.py

```diff
@@ -0,0 +1,397 @@
+"""
+Created on 29.04.2021
+
+@author: Filip Maciejewski
+@contact: filip.b.maciejewski@gmail.com
+"""
+
+import copy
+import numpy as np
+from typing import Optional, Dict
+from tqdm import tqdm
+from QREM import ancillary_functions as anf
+from ..child_classes.ddot_marginal_analyzer_vanilla import DDOTMarginalsAnalyzer
+from DDOT_module.child_classes.global_noise_matrix_creator import GlobalNoiseMatrixCreator
+
+
+class CorrectionDataGenerator(DDOTMarginalsAnalyzer):
+    """
+        1
+    """
+
+    def __init__(self,
+                 results_dictionary_ddot: dict,
+                 reverse_counts: bool,
+                 number_of_qubits: int,
+                 marginals_dictionary: dict,
+                 clusters_list: list,
+                 neighborhoods: dict,
+                 noise_matrices_dictionary: Optional[dict] = None
+                 ) -> None:
+
+        super().__init__(results_dictionary_ddot,
+                         reverse_counts,
+                         marginals_dictionary,
+                         noise_matrices_dictionary
+                         )
+        self._number_of_qubits = number_of_qubits
+        self._qubit_indices = range(number_of_qubits)
+
+        self._clusters_list = clusters_list
+        self._neighborhoods = neighborhoods
+
+        clusters_dictionary = {'q%s' % qi: [] for qi in self._qubit_indices}
+        for cluster in clusters_list:
```

Comment:
It's common practice to avoid nesting larger than 3 levels. Is there no way to reduce nesting here?